packages/cache/src/service.ts

```diff
@@ -0,0 +1,146 @@
+import { logger } from "@formbricks/logger";
+import type { RedisClient } from "../types/client";
+import { type CacheError, ErrorCode, type Result, err, ok } from "../types/error";
+import type { CacheKey } from "../types/keys";
+import { ZCacheKey, ZTtlMs, validateInputs } from "../types/service";
+
+/**
+ * Core cache service providing basic Redis operations with JSON serialization
+ */
+export class CacheService {
+  constructor(private readonly redis: RedisClient) {}
+
+  /**
+   * Get a value from cache with automatic JSON deserialization
+   * @param key - Cache key to retrieve
+   * @returns Result containing parsed value, null if not found, or an error
+   */
+  async get<T>(key: CacheKey): Promise<Result<T | null, CacheError>> {
+    const validation = validateInputs([key, ZCacheKey]);
```

Comment:
I think this is where the different error interfaces would make sense, this the `validateInputs` returns `CacheError`, the error code when checked against typescript does not error when comparing it with `ErrorCode.CacheCorruptionError` for example.
If the function only returned a `ValidationError`, TS would give instant feedback on this case

But maybe this is a very specific use case for it so I'm not sure, what do you think?

With `CacheError`:
<img width="612" height="80" alt="Image" src="https://github.com/user-attachments/assets/5f750054-dc02-4abb-b63c-cdbc07496fd4" />

With specific `ValidationError`
<img width="673" height="89" alt="Image" src="https://github.com/user-attachments/assets/3797c830-6ff6-4ab1-af93-14d41143845e" />